lib/src/ThemeContext.tsx

```diff
@@ -274,14 +274,28 @@ const parseTheme = (theme) => {
   return componentTokensCopy;
 };
 
-const ThemeProvider = ({ theme, advancedTheme, children }) => {
-  const parsedTheme = useMemo(
-    () => (theme && parseTheme(theme)) || (advancedTheme && parseAdvancedTheme(advancedTheme)),
-    [theme, advancedTheme]
-  );
+type ThemeProviderCommonProps = {
+  children: React.ReactNode;
+};
+type ThemeProviderDefaultTheme = ThemeProviderCommonProps & {
+  theme?: object;
+  advancedTheme?: never;
+};
+type ThemeProviderAdvancedTheme = ThemeProviderCommonProps & {
+  theme?: never;
```

Comment:
I don't understand the use of `never` type.

Reply:
`never` is a special type used to represent impossible combinations. In this case, I'm using it to guarantee that an advanced theme is never passed at the same time as a default theme, and the same in the opposite case. For example:

```
type ThemeProviderDefaultTheme = ThemeProviderCommonProps & {
  theme?: object;
  advancedTheme?: never;
};
```

This type represents the situation when the ThemeProvider receives a regular `theme`. By giving `advancedTheme` the type `never` I'm saying that this prop should be received in this case.

Reply:
And why did you discard the first solution? Is it not the same?
```
type ThemeProviderDefaultTheme = ThemeProviderCommonProps & {
  theme: object;
};
```